microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+import time
+from datetime import date
+
+from assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    search_space,
+)
+from lisa.base_tools import Cat, Uname
+from lisa.features import StartStop
+from lisa.operating_system import Redhat, Suse
+from lisa.sut_orchestrator.azure.tools import VmGeneration
+from lisa.testsuite import simple_requirement
+from lisa.tools import Blkid, Dmesg, Echo, Fdisk, Sed, Swap, SwapOn
+from lisa.util import (
+    LisaException,
+    SkippedException,
+    find_patterns_in_lines,
+    get_matched_str,
+)
+from lisa.util.logger import Logger
+from lisa.util.shell import wait_tcp_port_ready
+
+# matching 4-digit year between whitespace
+# or at the beginning of a line
+# example: 2022 from '2022-02-04 11:38:06.949136+0000'
+# or 2018 from 'Fri 07 Sep 2018 11:26:52 AM CEST .838868 seconds'
+# will be matched
+_YEAR_PATTERN = re.compile("(\s\d{4}\s)|(^\d{4})")
```

Comment:
Try this to capture the 4 digits of year only.

`(?:\s(\d{4})\s)|(?:^(\d{4}))`

Reply:
using '|' will make get_matched_str return a tuple not a str because it's actually using 're.findall'. 
So I'll change to use re.search in get_year function. Or do you think it's better to add a new function (e.g. get_first_matched_str) in init.py?

Reply:
Try this one? It captures the first 4 digits

`^(?:[\w\W]*?)(\d{4})(?:[-\s])`